### PR TITLE
Release v1.4.3

### PR DESCRIPTION
**Pull request by beorn7**
This release merely updates dependencies and, as a byproduct, the binaries that come with the release will be built with the current Go version (1.18.x).

The last release is quite long ago, and a lot of security issues have accumulated in depencies and the Go version used to build the binaries of the previous release. I do not now of any issues relevant to the Pushgateway, but at this point, it's probably better to cut a new release to avoid false alerts (and maybe even avoid an issue we don't know about yet). It comes with the risk of pulling in new bugs, though.

### CODE REVIEW

**Comment by beorn7** [comment]
Sorry for weekend PR. I set the release date to Monday. Feel free to review when the weekend is over.

**Comment by beorn7** [comment]
Fixes #469.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,11 +1,12 @@
+---
 version: 2.1
 orbs:
   prometheus: prometheus/prometheus@0.16.0
 executors:
   # Whenever the Go version is updated here, .promu.yml should also be updated.
   golang:
     docker:
-      - image: circleci/golang:1.17
+      - image: quay.io/prometheus/golang-builder:1.18-base
```
Comment (SuperQ) [inline .circleci/config.yml:9]: We're actually switching these over to `cimg/go:1.18` except for `prometheus/prometheus`.

**Review by SuperQ — CHANGES_REQUESTED** [review]
I would recommend changing the minimum Go version in go.mod to 1.17 and running a go mod tidy to get the new indirect dependency format.

**Comment by beorn7** [comment]
Both comments addressed. Thank you very much. PHAL.

**Review by SuperQ — APPROVED** [review]
LGTM

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**